### PR TITLE
Clientary actions & sources

### PR DESCRIPTION
**Pull request by feyzullah**
### Actions
* Create Client
* Create Expense
* Create Estimate
* Create Invoice
* Create Task
### Sources
* New Contact Created
* New Expense Created
* New Payment Created
* New Project Created
* New Staff Created
* New Task Created
* New Invoice Created
* New Client Created
* New Estimate Created

### CODE REVIEW

File: components/clientary/app/clientary.app.ts [hunk 1]
```diff
@@ -1,13 +1,114 @@
...
+    },
+    projectId: {
+      type: "integer",
+      label: "Project ID",
+      description: "Project ID",
+      async options({ page }) {
+        page++;
+        return await utils.getAsyncOptions({
+          resourceFn: this.getRequestMethod("getProjects"),
+          page,
```
Comment (lcaresia) [inline components/clientary/app/clientary.app.ts:36]:
```suggestion
        return await utils.getAsyncOptions({
          resourceFn: this.getRequestMethod("getProjects"),
          page: page + 1,
```

File: components/clientary/app/clientary.app.ts [hunk 2]
```diff
@@ -1,13 +1,114 @@
...
+    },
+    invoiceId: {
+      type: "integer",
+      label: "Invoice ID",
+      description: "Invoice ID",
+      async options({ page }) {
+        page++;
+        return await utils.getAsyncOptions({
+          resourceFn: this.getRequestMethod("getInvoices"),
+          page,
```
Comment (lcaresia) [inline components/clientary/app/clientary.app.ts:51]:
```suggestion
        return await utils.getAsyncOptions({
          resourceFn: this.getRequestMethod("getInvoices"),
          page: page + 1,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Just some fixes!

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
@feyzullah Please check the following improvements,

1. Create Estimate Action -`Estimate Item Attributes` should be a mandatory prop
image


2. Create Expense Action - `client_id` and `invoice_id` is not properly passed to Clientary
image


3. Create Invoice Action - `Estimate Item Attributes` should be a mandatory prop
image


4. Create Task Action - `client_id`  is not properly passed to Clientary
image

**Comment by feyzullah** [comment]
> @feyzullah Please check the following improvements,
> 
> 1. Create Estimate Action -`Estimate Item Attributes` should be a mandatory prop
>    image
> 2. Create Expense Action - `client_id` and `invoice_id` is not properly passed to Clientary
>    image
> 3. Create Invoice Action - `Estimate Item Attributes` should be a mandatory prop
>    image
> 4. Create Task Action - `client_id`  is not properly passed to Clientary
>    image

Hi @DilanAthukorala I have fixed 1st and 3rd ones. They are still optional but you won't see that error if they are not given.
For 2nd and 4th ones, if `client_id` is given with the `project_id`, the project must have the related client in clients on Clientary.
In 2nd one, I was not able to pass `invoice_id` to Clientary under any condition, so I removed that prop from the action. Can you please check once more?

**Comment by DilanAthukorala** [comment]
Verification completed, ready for release!

**Review by alysonturing — APPROVED** [review]
LGTM!

**Pull request merged**